bench/pmc_tool.c

```
@@ -0,0 +1,245 @@
+        fprintf(stderr, "kpep_db_create failed: %d\n", ret);
+        return 1;
+    }
+
+    if (strcmp(argv[1], "list") == 0) {
+        size_t count = 0;
+        kpep_db_events_count(db, &count);
+        kpep_event **evs = malloc(count * sizeof(void *));
+        kpep_db_events(db, evs, count * sizeof(void *));
+        const char *filter = argc > 2 ? argv[2] : NULL;
+        for (size_t i = 0; i < count; i++) {
+            if (filter && !strcasestr(evs[i]->name, filter)) continue;
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)